doc/lsst.pipe.base/task-framework-overview.rst

```diff
@@ -0,0 +1,47 @@
+.. _task-framework-overview:
+
+##############################
+Overview of the task framework
+##############################
+
+:doc:`lsst.pipe.base <index>` provides a data processing pipeline infrastructure.
+Data processing is performed by **tasks**, which are instances of `lsst.pipe.base.Task` or `lsst.pipe.base.CmdLineTask`.
+Tasks perform a wide range of data processing operations, from basic operations such as assembling raw images into CCD images (trimming overscan), fitting a WCS or detecting sources on an image, to complex combinations of such operations.
+
+**Tasks** are hierarchical.
+Each task may may call other tasks to perform some of its data processing; we say that a **parent task** calls a **subtask**.
+The highest-level task is called the **top-level task**.
+To call a subtask, the parent task constructs the subtask and then calls methods on it.
+Thus data transfer between tasks is simply a matter of passing the data as arguments to function calls.
+
+**Command-line tasks** are tasks that can be run from the command line.
+You might think of them as the LSST equivalent of a data processing pipeline.
+Despite their extra capabilities, command-line tasks can also be used as ordinary tasks and called as subtasks by other tasks.
+Command-line tasks are subclasses of `~lsst.pipe.base.CmdLineTask`.
+
+Each task is configured using the `lsst.pex.config` module, using a task-specific subclass of `lsst.pex.config.Config`.
+The task's configuration includes all subtasks that the task may call.
+As a result, it is easy to replace (or "retarget") one subtask with another.
+A common use for this is to provide a camera-specific variant of a particular task, e.g. use one version for SDSS imager data and another version for Subaru Hyper Suprime-Cam data).
+
+Tasks may process multiple items of data in parallel, using Python's `multiprocessing` library.
+Support for this is built into the `~lsst.pipe.base.ArgumentParser` and `~lsst.pipe.base.TaskRunner`.
+
+Most tasks have a ``run`` method that performs the primary data processing.
+Each task's `run` method should return a `~lsst.pipe.base.Struct`.
+This allows named access to returned data, which provides safer evolution than relying on the order of returned values.
+All task methods that return more than one or two items of data should return the data in a `~lsst.pipe.base.Struct`.
+
+Many tasks are found in the ``pipe_tasks`` package, especially tasks that use many different packages and don't seem to belong in any one of them.
```

Comment:
Might be clearer to refer to `pipe_tasks` as `lsst.pipe.tasks` instead/in addition to match the `lsst.pipe.base` format. Eventually it would link to equivalent docs! (Same goes for referring to `ip_isr` right after.) Or perhaps the general term should be `module_name` and only python specific things should be `lsst.module.name`? I think you understand the goal of consistency I'm getting at here.